File: Lista de Exercícios/Go/12. Métodos de Conveniência/MetodosConveniencia.go
package main

import (
	"fmt"
)

type Produto struct {
	nomeProduto   string
	valorProduto  float64
}

func (p Produto) adicionar(outro Produto) Produto {
	return Produto{
		nomeProduto:  "O resultado da soma de valor dos produtos é:",
		valorProduto: p.valorProduto + outro.valorProduto,
	}
}

func (p Produto) String() string {
	return fmt.Sprintf("%s: R$%.2f", p.nomeProduto, p.valorProduto)
}

func main() {

}

File: Lista de Exercícios/Java/12. Métodos de Conveniência/MetodosConveniencia.java
class Produto {
    private String nomeProduto;
    private double valorProduto;

    public Produto(String nomeProduto, double valorProduto) {
        this.nomeProduto = nomeProduto;
        this.valorProduto = valorProduto;
    }

    public Produto somar(Produto outro) {
        if (outro != null) {
            double valorTotal = this.valorProduto + outro.valorProduto;
            return new Produto("O resultado da soma de valor dos produtos é: R$", valorTotal);
        }
        return null;
    }

    @Override
    public String toString() {
        return nomeProduto + ": R$" + valorProduto;
    }

    public static void main(String[] args) {

    }
}